Core/Pixel.swift

```diff
@@ -305,9 +307,11 @@ public class TimedPixel {
         self.date = date
     }
     
-    public func fire(_ fireDate: Date = Date()) {
+    public func fire(_ fireDate: Date = Date(), withAdditionalParmaeters params: [String: String?] = [:]) {
         let duration = String(fireDate.timeIntervalSince(date))
-        Pixel.fire(pixel: pixel, withAdditionalParameters: [PixelParameters.duration: duration])
+        var newParams = params
+        newParams[PixelParameters.duration] = duration
```

Comment:
Maybe we should ensure that `params` dict already does not contain `duration` key before assigning it (to avoid stupid mistakes :) )?

Reply:
I think it would be unlock to do it by accident and **should** be picked up when testing anyway. :)